process/nettisa.hpp

```diff
@@ -0,0 +1,201 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (C) 2023, CESNET z.s.p.o.
+ */
+
+/**
+ * \file nettisa.hpp
+ * \brief Class for creating NetTiSA flow.
+ * \author Josef Koumar koumajos@fit.cvut.cz
+ * \date 2023
+ */
+
+#ifndef IPXP_PROCESS_NETTISA_HPP
+#define IPXP_PROCESS_NETTISA_HPP
+
+#include <cstring>
+#include <vector>
+#include <utility>
+#include <math.h> 
+
+#ifdef WITH_NEMEA
+  #include "fields.h"
+#endif
+
+#include <ipfixprobe/process.hpp>
+#include <ipfixprobe/flowifc.hpp>
+#include <ipfixprobe/packet.hpp>
+#include <ipfixprobe/ipfix-elements.hpp>
+
+namespace ipxp {
+
+#define NETTISA_UNIREC_TEMPLATE "NTS_MEAN,NTS_MIN,NTS_MAX,NTS_STDEV,NTS_KURTOSIS,NTS_ROOT_MEAN_SQUARE,NTS_AVERAGE_DISPERSION,NTS_MEAN_SCALED_TIME,NTS_MEAN_DIFFTIMES,NTS_MIN_DIFFTIMES,NTS_MAX_DIFFTIMES,NTS_TIME_DISTRIBUTION,NTS_SWITCHING_RATIO"
+
+UR_FIELDS (
+   float NTS_MEAN,
+   float NTS_MIN,
+   float NTS_MAX,
+   float NTS_STDEV,
+   float NTS_KURTOSIS,
+   float NTS_ROOT_MEAN_SQUARE,
+   float NTS_AVERAGE_DISPERSION,
+   float NTS_MEAN_SCALED_TIME,
+   float NTS_MEAN_DIFFTIMES,
+   float NTS_MIN_DIFFTIMES,
+   float NTS_MAX_DIFFTIMES,
+   float NTS_TIME_DISTRIBUTION,
+   float NTS_SWITCHING_RATIO,
+)
+
+/**
+ * \brief Flow record extension header for storing parsed NETTISA data.
+ */
+struct RecordExtNETTISA : public RecordExt {
+   static int REGISTERED_ID;
+
+   float mean;
+   float min;
+   float max;
+   float stdev;
+   float kurtosis;
+   float root_mean_square;
+   float average_dispersion;
+   float mean_scaled_time;
+   float mean_difftimes;
+   float min_difftimes;
+   float max_difftimes;
+   float time_distribution;
+   float switching_ratio;
+
+   uint16_t prev_payload;
+   long prev_time;
+    
+   RecordExtNETTISA() : RecordExt(REGISTERED_ID)
+   {
+       mean = 0;
+       min = 0;
```

Comment:
Nemelo by minimum byt nastaveno na nejvyssi moznou hodnotu, to stejne u `min_difftimes`

Reply:
Dal bych tam maximalni moznou hodnotu:
`min = std::numeric_limits<float>::max();`